### ISSUE TITLE
1.4.55 fails to build (std::function doesn't name a template type)

### ISSUE THREAD

**Issue by Sleepyowl** [aws/aws-sdk-cpp#881]
I'm building with gcc-7.3.0 and build fails with
```
aws-cpp-sdk-core/include/aws/core/utils/ratelimiter/DefaultRateLimiter.h:42:54: error: ‘function’ in namespace ‘std’ does not name a template type
                 using ElapsedTimeFunctionType = std::function< InternalTimePointType() >;
```
Adding `#include <functional>` to the header solves the issue. (Sorry for not making a PR for this.)

**Comment by marcomagdy** [comment]
There's already a PR. https://github.com/aws/aws-sdk-cpp/pull/880 and you answered my question on it. I'll be merging it ASAP. Thanks

**Comment by marcomagdy** [comment]
Resolved in 1.4.56
